flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemTableSink.java

```diff
@@ -374,7 +374,8 @@ public DynamicTableSource.DataStructureConverter createDataStructureConverter(
     }
 
     private Path toStagingPath() {
-        Path stagingDir = new Path(path, ".staging_" + System.currentTimeMillis());
+        Path stagingDir =
+                new Path(path, ".staging_" + UUID.randomUUID() + System.currentTimeMillis());
```

Comment:
nit: do you thinkt it is worth to add a comment why we need it?
May be not the full description here, however e.g. a link to the corresonding comment in jira?

Reply:
I guess, we don't need the `System.currentMillis()` if we use UUIDs, do we?